storage/badger/operation/modifiers.go

```diff
@@ -42,3 +44,15 @@ func RetryOnConflictTx(db *badger.DB, action func(*badger.DB, func(*transaction.
 		return err
 	}
 }
+
+// TerminateOnFullDisk helper function to crash node if write failed because disk is full
+func TerminateOnFullDisk(err error) error {
+	// using panic so any deferred functions can still execute
+	// string check on Badger's or other Wraps on syscall disk error, instead of Unwrap or Is.
```

Comment:
`Is` should match any error in the chain, so I don't think you need to compare strings here: https://pkg.go.dev/errors#Is